gtda/images/filtrations.py

```diff
@@ -349,3 +350,135 @@ def transform(self, X, y=None):
+    its closest activated pixel. The dilation process is iterative and the
+    number of iterations `n_iterations` sets the maximum distance considered.
+    Deactivated pixels further away from activated pixels are assigned the
+    maximum Manhattan distance between two pixels on the image and activated
+    pixels are assigned a value of 0.
+
+    Parameters
+    ----------
+    n_iterations : int or None, optional, default: ``None``
+        Number of iterations in the dilation process. ``None`` means dilation
+        over the full image.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    n_iterations_ : int
+        Effective number of iterations in the dilation process. Set in
+        :meth:`fit`.
+
+    max_value_: float
+        Maximum pixel value among all pixels in all images of the collection.
+        Set in :meth:`fit`.
+
+    See also
+    --------
+    gtda.homology.CubicalPersistence, Binarizer
+
+    """
+    _hyperparameters = {'n_iterations_': [int, (1, np.inf)]}
+
+    def __init__(self, n_iterations=None, n_jobs=None):
+        self.n_iterations = n_iterations
+        self.n_jobs = n_jobs
+
+    def _calculate_dilation(self, X):
+        Xd = X * 1.
+
+        for iteration in range(1, min(self.n_iterations_,
+                                      self.max_value_) + 1):
+            Xtemp = np.asarray([ndi.binary_dilation(Xd[i])
+                                for i in range(Xd.shape[0])])
+            Xd += (Xd + Xtemp == 1) * Xtemp * (iteration + 1)
+
+        mask_filtered = Xd == 0
+        Xd -= 1
+        Xd[mask_filtered] = self.max_value_
+        return Xd
+
+    def fit(self, X, y=None):
+        """Calculate `n_iterations_` and 'max_value_'from a collection of
+        binary images. Then, return the estimator.
+
+        This method is here to implement the usual scikit-learn API and hence
+        work in pipelines.
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_pixels_x, n_pixels_y [, n_pixels_z])
+            Input data. Each entry along axis 0 is interpreted as a 2D or 3D
+            binary image.
+
+        y : None
+            There is no need of a target in a transformer, yet the pipeline API
+            requires this parameter.
+
+        Returns
+        -------
+        self : object
+
+        """
+        X = check_array(X,  ensure_2d=False, allow_nd=True)
+
+        if self.n_iterations is None:
+            self.n_iterations_ = int(np.max(X.shape[1:]))
+        else:
+            self.n_iterations_ = self.n_iterations
+
+        validate_params({**self.get_params(),
+                         'n_iterations_': self.n_iterations_},
+                        self._hyperparameters)
+
+        self.max_value_ = np.max(X.shape[1:])
```

Comment:
This line could be placed above the definition of `self.n_iterations_` and then used in the first `if` branch.